pgml-cms/docs/guides/unified-rag.md

```diff
@@ -0,0 +1,332 @@
+  3 | 0.07361361603777417 | Here is another example of using the pgml.transform function
+  6 |  0.1615426790254476 | SELECT pgml.transform(                                      +
+    |                     |   task   => '{                                              +
+    |                     |     "task": "text-generation",                              +
+    |                     |     "model": "microsoft/Phi-3-mini-128k-instruct"           +
+    |                     |   }'::JSONB,                                                +
+    |                     |   inputs  => ARRAY['AI is going to'],                       +
+    |                     |   args   => '{                                              +
+    |                     |     "max_new_tokens": 100                                   +
+    |                     |   }'::JSONB                                                 +
+    |                     | );
+  2 | 0.16822049523056992 | SELECT pgml.transform(                                      +
+    |                     |   task   => '{                                              +
+    |                     |     "task": "text-generation",                              +
+    |                     |     "model": "meta-llama/Meta-Llama-3-8B-Instruct"          +
+    |                     |   }'::JSONB,                                                +
+    |                     |   inputs  => ARRAY['AI is going to'],                       +
+    |                     |   args   => '{                                              +
+    |                     |     "max_new_tokens": 100                                   +
+    |                     |   }'::JSONB                                                 +
+    |                     | );
+  4 | 0.16937527108941763 | SELECT pgml.transform(                                      +
+    |                     |   task   => '{                                              +
+    |                     |     "task": "text-generation",                              +
+    |                     |     "model": "meta-llama/Meta-Llama-3-70B-Instruct"         +
+    |                     |   }'::JSONB,                                                +
+    |                     |   inputs  => ARRAY['AI is going to'],                       +
+    |                     |   args   => '{                                              +
+    |                     |     "max_new_tokens": 100                                   +
+    |                     |   }'::JSONB                                                 +
+    |                     | );
+(6 rows)
+
+!!!
+
+!!!
+
+We are using a CTE to embed the user query, and then performing nearest neighbors search using the cosine similarity function to compare the distance between our embeddings. Note how fast this is! Our embeddings utilize an HNSW index from pgvector to perform ridiculously fast retrieval.
```

Comment:
What's `CTE` stand for? 🤔

Reply:
It stands for common table expression: https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-cte/